js/src/dropdown.js

```diff
@@ -289,6 +289,24 @@ class Dropdown extends BaseComponent {
     return this._element.closest(`.${CLASS_NAME_NAVBAR}`) !== null
   }
 
+  _getOffset() {
+    let { offset } = this._config
+
+    if (!offset) {
+      return [0, 0]
+    }
+
+    if (typeof offset === 'string') {
+      offset = offset.split(',').map(val => Number.parseInt(val, 10))
+    } else if (typeof offset === 'function') {
+      offset = ({ popper, reference, placement }) => {
+        return offset({ popper, reference, placement }, this._element)
```

Comment:
Destructuring here might be ignored because destructuring is not being used in real, thanks for considering previous comment :grin:

---
Popper may call the `offset` function multiple times, and because the `offset` function is in the closure I am getting the error: **Maximum call stack size exceeded** (You can check by creating dist files locally and passing a function to the offset in dropdowns configuration)

My suggestion:
```js
if (typeof offset === 'string') {
  return offset.split(',').map(val => Number.parseInt(val, 10))
}

if (typeof offset === 'function') {
  return popperData => {
    return offset(popperData, this._element)
  }
}
```

Reply:
I found this "strange" behaviour in js:
```
let x = () => 1
x() // print 1
x = () => x()
x() // infinite recursion
```
so I've to rewrite the variable name in this script.